client/src/pages/layout/Layout.js

```diff
@@ -1,19 +1,25 @@
 import React, { useEffect } from 'react';
 import { Redirect } from 'react-router-dom';
 import { LinearProgress } from '@material-ui/core';
-
 import Navbar from './Navbar';
 import * as actions from '../../context/actions';
-import { useGlobalContext } from '../../context/studyappContext';
+import { useSocketContext, useGlobalContext } from '../../context/studyappContext';
 
 const Layout = ({ children }) => {
     const { dispatch, isLoading, isAuth } = useGlobalContext();
-
+    const { SocketManager } = useSocketContext();
     useEffect(() => {
         actions.fetchProfile()(dispatch);
     }, [dispatch]);
 
-    if (!isAuth) return <Redirect to="/login" />;
+    if (isAuth)
+        SocketManager.startSocket();
+
+    if (!isAuth) {
+        SocketManager.closeSocket();
+        return <Redirect to="/login" />
+    };
```

Comment:
If the user refreshes the page (as opposed to logging out), this won't run. So the socket won't be properly disconnected  and next time it connects, you might have issues.

Reply:
Updated so closeSocket is called when Layout mounts/unmounts.